pkg/api/folder.go

```diff
@@ -84,8 +86,22 @@ func UpdateFolder(c *models.ReqContext, cmd models.UpdateFolderCommand) response
 	return response.JSON(200, toFolderDto(g, cmd.Result))
 }
 
-func DeleteFolder(c *models.ReqContext) response.Response {
+func (hs *HTTPServer) DeleteFolder(c *models.ReqContext) response.Response { // temporary adding this function to HTTPServer, will be removed from HTTPServer when librarypanels featuretoggle is removed
 	s := dashboards.NewFolderService(c.OrgId, c.SignedInUser)
+	if hs.Cfg.IsPanelLibraryEnabled() {
+		f, err := s.GetFolderByUID(c.Params(":uid"))
+		if err != nil {
+			return toFolderError(err)
+		}
+		err = hs.LibraryPanelService.DeleteLibraryPanelsInFolder(c, f)
+		if errors.Is(err, librarypanels.ErrFolderHasConnectedLibraryPanels) {
+			return response.Error(500, "Folder could not be deleted because it contains linked library panels", err)
```

Comment:
Should this really be a 500? Isn't it a user error rather than an internal server error?

Reply:
yea, this 403 Forbidden feels like the most appropriate status code 
https://en.wikipedia.org/wiki/HTTP_403

Reply:
I agree, I'll fix it.

Reply:
Fixed!